lib/ansible/modules/cloud/amazon/aws_codebuild.py

```diff
@@ -0,0 +1,421 @@
+        # and more... see http://boto3.readthedocs.io/en/latest/reference/services/codebuild.html#CodeBuild.Client.create_project
+    cache:
+      description: Cache settings for the build project.
+      returned: when configured
+      type: complex
+    environment:
+      description: Environment settings for the build
+      returned: always
+      type: complex
+    service_role:
+      description: IAM role to be used during build to access other AWS services.
+      returned: always
+      type: string
+      sample: arn:aws:iam::123123123:role/codebuild-service-role
+    timeout_in_minutes:
+      description: The timeout of a build in minutes
+      returned: always
+      type: int
+      sample: 60
+    tags:
+      description: Tags added to the project
+      returned: when configured
+      type: list
+    created:
+      description: Timestamp of the create time of the project
+      returned: always
+      type: string
+      sample: 2018-04-17T16:56:03.245000+02:00
+'''
+
+import traceback
+from ansible.module_utils._text import to_native
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, snake_dict_to_camel_dict
+
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def create_or_update_project(client, params, module):
+    resp = {}
+    name = params['name']
+    # Sanity check and cleanup params not needed for boto methods:
+    if not isinstance(name, str):
+        module.fail_json(msg="Params was missing name", exception=traceback.format_exc())
+    clean_params = dict((k, v) for k, v in params.items() if v is not None)
+    clean_params.pop('region', None)
+    clean_params.pop('state', None)
+    clean_params.pop('validate_certs', None)
+    formatted_params = snake_dict_to_camel_dict(clean_params)
+
+    # Check if project with that name aleady exists and if so update existing:
+    found = describe_project(client=client, name=name, module=module)
+    changed = False
+
+    if 'name' in found:
+        found_project = found
+        resp = update_project(client=client, params=formatted_params, module=module)
+        updated_project = resp['project']
+
+        # Prep both dicts for sensible change comparison:
+        found_project.pop('lastModified')
+        updated_project.pop('lastModified')
+        if 'tags' not in updated_project:
+            updated_project['tags'] = []
+
+        if updated_project != found_project:
+            changed = True
+        return resp, changed
+    # Or create new project:
+    try:
+        resp = client.create_project(**formatted_params)
+        changed = True
+        return resp, changed
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable create project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
```

Comment:
This (and all of the other exception handling) should be using module.fail_json_aws() instead of doing manual exception formatting.